grammar/ja/src/rules.rs

```diff
@@ -1274,6 +1277,131 @@ pub fn rules_time(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()> {
+                     .span_to(&helpers::hour(19, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+    b.rule_2("afternoon <time-of-day>",
+        b.reg(r#"昼の?"#)?,
+        time_of_day_check_hour!(1, 7, 13, 19),
+        |_, tod| {
+            let period = helpers::hour(13, false)?
+                     .span_to(&helpers::hour(19, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+
+    b.rule_2("evening <time-of-day>",
+        time_check!(form!(Form::PartOfDay(PartOfDayForm::Evening))),
+        time_of_day_check_hour!(7, 11, 19, 23),
+        |_, tod| {
+            let period = helpers::hour(19, false)?
+                     .span_to(&helpers::hour(24, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+
+    b.rule_3("evening <time-of-day>",
+        time_check!(form!(Form::PartOfDay(PartOfDayForm::Evening))),
+        b.reg(r#"の"#)?,
+        time_of_day_check_hour!(7, 11, 19, 23),
+        |_, _, tod| {
+            let period = helpers::hour(19, false)?
+                     .span_to(&helpers::hour(24, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+
+    b.rule_2("night <time-of-day>",
+        time_check!(form!(Form::PartOfDay(PartOfDayForm::Night))),
+        time_of_day_check_hour!(0, 4),
+        |_, tod| {
+            let period = helpers::hour(0, false)?
+                     .span_to(&helpers::hour(4, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+
+    b.rule_3("night <time-of-day>",
+        time_check!(form!(Form::PartOfDay(PartOfDayForm::Night))),
+        b.reg(r#"の"#)?,
+        time_of_day_check_hour!(0, 4),
+        |_, _, tod| {
+            let period = helpers::hour(0, false)?
+                     .span_to(&helpers::hour(4, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+
+    b.rule_2("night <time-of-day>",
+        time_check!(form!(Form::PartOfDay(PartOfDayForm::Night))),
+        time_of_day_check_hour!(7, 11, 19, 23),
+        |_, tod| {
+            let period = helpers::hour(19, false)?
+                     .span_to(&helpers::hour(0, false)?, true)?;
+            Ok(tod.value().intersect(&period)?
+                .form(tod.value().form.clone()))
+        }
+    );
+
+    b.rule_3("night <time-of-day>",
+        time_check!(form!(Form::PartOfDay(PartOfDayForm::Night))),
+        b.reg(r#"の"#)?,
+        time_of_day_check_hour!(7, 11, 19, 23),
+        |_, _, tod| {
+            let period = helpers::hour(19, false)?
+                     .span_to(&helpers::hour(0, false)?, true)?;
```

Comment:
Shouldn't it be 23 instead of 0?

Reply:
It should be 23 because 0 case is handled previously by 

```
    b.rule_2("night <time-of-day>",
        time_check!(form!(Form::PartOfDay(PartOfDayForm::Night))),
        time_of_day_check_hour!(0, 4),
        |_, tod| {
            let period = helpers::hour(0, false)?
                     .span_to(&helpers::hour(4, false)?, true)?;
            Ok(tod.value().intersect(&period)?
                .form(tod.value().form.clone()))
        }
    );
```